pkg/proxy/iptables/proxier.go

```diff
@@ -237,14 +238,8 @@ func (ecm *endpointsChangeMap) update(namespacedName *types.NamespacedName, prev
 	change.current = endpointsToEndpointsMap(current, ecm.hostname)
 	if reflect.DeepEqual(change.previous, change.current) {
 		delete(ecm.items, *namespacedName)
-		return false
-	}
-	// TODO: Instead of returning true/false, we should consider returning whether
-	// the map contains some element or not. Currently, if the change is
-	// "reverting" some previous endpoints update, but there are still some other
-	// modified endpoints, we will return false, even though there are some change
-	// to apply.
-	return true
+	}
+	return len(ecm.items) > 0
```

Comment:
I guess this works.  I was thinking of the return as "did this update() cause a change", but this should be behaviorally equivalent at the end.